converter.go

```diff
@@ -851,98 +936,16 @@ func arrowToValue(ctx context.Context, rowIdx int, srcColumnMeta execResponseRow
 				}
-			case *array.Int64:
-				return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (int64, error) {
-					return arrowInt64ToValue(typedValues, j, higherPrecision, srcColumnMeta.Fields[0].Scale).(int64), nil
-				})
-			case *array.Int32:
-				return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (int32, error) {
-					return arrowInt32ToValue(typedValues, j, higherPrecision, srcColumnMeta.Fields[0].Scale).(int32), nil
-				})
-			case *array.Int16:
-				return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (int16, error) {
-					return arrowInt16ToValue(typedValues, j, higherPrecision, srcColumnMeta.Fields[0].Scale).(int16), nil
-				})
-			case *array.Int8:
-				return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (int8, error) {
-					return arrowInt8ToValue(typedValues, j, higherPrecision, srcColumnMeta.Fields[0].Scale).(int8), nil
-				})
+				return buildStructuredArray(ctx, srcColumnMeta.Fields[0], arr, params)
 			}
-		case realType:
-			return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (float64, error) {
-				return arrowRealToValue(values.(*array.Float64), j).(float64), nil
-			})
-		case textType:
-			return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (string, error) {
-				return arrowStringToValue(values.(*array.String), j).(string), nil
-			})
-		case booleanType:
-			return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (bool, error) {
-				return arrowBoolToValue(values.(*array.Boolean), j).(bool), nil
-			})
-		case binaryType:
-			return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) ([]byte, error) {
-				return arrowBinaryToValue(values.(*array.Binary), j).([]byte), nil
-			})
-		case dateType:
-			return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (time.Time, error) {
-				return arrowDateToValue(values.(*array.Date32), int(j)).(time.Time), nil
-			})
-		case timeType:
-			return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (time.Time, error) {
-				return arrowTimeToValue(values, int(j), srcColumnMeta.Fields[0].Scale).(time.Time), nil
-			})
-		case timestampNtzType, timestampLtzType, timestampTzType:
-			return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (time.Time, error) {
-				ptr := arrowSnowflakeTimestampToTime(values, snowflakeType, srcColumnMeta.Fields[0].Scale, int(j), loc)
-				if ptr != nil {
-					return *ptr, nil
-				}
-				return time.Time{}, nil
-			})
-		case objectType:
-			return mapStructuredArrayNativeArrowRows(offsets, rowIdx, func(j int) (*structuredType, error) {
-				m := make(map[string]any, len(srcColumnMeta.Fields[0].Fields))
-				for fieldIdx, field := range srcColumnMeta.Fields[0].Fields {
-					m[field.Name] = values.(*array.Struct).Field(fieldIdx).ValueStr(int(j))
-				}
-				return buildStructuredTypeRecursive(m, srcColumnMeta.Fields[0].Fields, params), nil
-			})
+			return nil, nil
```

Comment:
perhaps we can log that this ocurred?

Reply:
Why? This is perfectly normal situation, that we have null value from DB.